codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -166,18 +171,89 @@ private void constructHandler() {
      * Adds middleware to the operation middleware stack.
      */
     private void populateOperationMiddlewareStack() {
+
+        // Populate middleware's from runtime client plugins
         runtimeClientPlugins.forEach(runtimeClientPlugin -> {
             if (!runtimeClientPlugin.matchesService(model, service)
                     && !runtimeClientPlugin.matchesOperation(model, service, operation)) {
                 return;
             }
 
-            if (runtimeClientPlugin.buildMiddlewareStack().isPresent()) {
-                runtimeClientPlugin.buildMiddlewareStack().get().applyMiddleware(
-                        writer, service, operation, protocolGenerator, "stack"
-                );
+            if (runtimeClientPlugin.registerMiddleware().isPresent()) {
+                MiddlewareRegistrar middlewareRegistrar = runtimeClientPlugin.registerMiddleware().get();
+                Collection<Symbol> functionArguments = middlewareRegistrar.getFunctionArguments();
+
+                if (middlewareRegistrar.getInlineRegisterMiddlewareStatement() != null
+                        && middlewareRegistrar.getInlineRegisterMiddlewarePosition() != null) {
+                    String registerStatement = String.format("stack.%s",
+                            middlewareRegistrar.getInlineRegisterMiddlewareStatement());
+                    writer.writeInline(registerStatement);
+                    writer.writeInline("$T(", middlewareRegistrar.getResolvedFunction());
+                    if (functionArguments != null) {
+                        List<Symbol> args = new ArrayList<>(functionArguments);
+                        for (Symbol arg: args) {
+                            writer.writeInline("$P", arg);
+                            if (args.iterator().hasNext()) {
```

Comment:
Have you tested this bit out? Not sure if it'll work

Reply:
Yes I did test it out, it worked.

Reply:
It worked. But worked because of `GoFmt` cleared the last `,`. Don't need this if check.